### PR TITLE
[Raymath] Add C++ operator overloads for common math function

### PR DESCRIPTION
**Pull request by JeffM2501**
This PR adds operator overloads for C++ for the math operations that have raymath functions.
Vectror2, 3, 4, and Quaternion are supported.
All operators are defined inline, and the define RAYLIB_DISABLE_VECTOR_OPERATORS can be used to disable this functionality.

While raylib is C, and not C++ and normally C++ code should be kept out of raylib, these operators offer exceptional utility and thus are an exception.

### CODE REVIEW

**Comment by JeffM2501** [comment]
For credit, this idea was from @Peter0x44 initally.

**Comment by Peter0x44** [comment]
Actually I got the idea from box2d which does the same thing.

**Comment by Peter0x44** [comment]
Should the operators call raymath functions or should they just be inlined? There is a policy that the functions don't call each other, not sure if it should be extended to these operators or not.

**Comment by JeffM2501** [comment]
I very specifically had them call the raymath functions so that we did not duplicate code. If a change has to be made to some math function, I don't want that change to have to be made in two places. The goal of this is to minimize maintenance cost.

File: src/raymath.h [hunk 1]
```diff
@@ -2580,4 +2580,319 @@ RMAPI void MatrixDecompose(Matrix mat, Vector3 *translation, Quaternion *rotatio
     }
 }
 
+
+// optional C++ math operators
+// define RAYLIB_DISABLE_VECTOR_OPERATORS to disable
+#if defined(__cplusplus) && !defined(RAYLIB_DISABLE_VECTOR_OPERATORS)
```
Comment (Peter0x44) [inline src/raymath.h:2586]: The operators aren't just for vectors, and raymath can be used separately from raylib.
Probably this should just be named RAYMATH_DISABLE_OPERATORS or RAYMATH_NO_OPERATORS.

**Comment by Peter0x44** [comment]
I would say most of the functions the operators call are not subject to change so incoming them might be better to stay consistent. But that's up to Tay.

File: src/raymath.h [hunk 2]
```diff
@@ -2580,4 +2582,319 @@ RMAPI void MatrixDecompose(Matrix mat, Vector3 *translation, Quaternion *rotatio
...
+
+inline const Quaternion& operator -= (Quaternion& lhs, const float& rhs)
+{
+	lhs = QuaternionSubtractValue(lhs, rhs);
+	return lhs;
+}
+
+inline Quaternion operator * (const Quaternion& lhs, const Matrix& rhs)
+{
+	return QuaternionTransform(lhs, rhs);
```
Comment (raysan5) [inline src/raymath.h:2890]: Please, could you replace TABs by 4 spaces?

**Comment by raysan5** [comment]
@JeffM2501 @Peter0x44 Thanks for the improvement, it looks good to me, let me know if it is ready to be merged!

**Comment by raysan5** [comment]
@JeffM2501 @Peter0x44 thanks for the improvement! merged! I'm reviewing the formating and details myself!

**Comment by raysan5** [comment]
@JeffM2501 @Peter0x44 Just noticed there are no operators for `Matrix`, should there be some added?

**Comment by JeffM2501** [comment]
> @JeffM2501 @Peter0x44 Just noticed there are no operators for `Matrix`, should there be some added?
I just added some matrix operators. I'm not sure they are super important.

**Comment by JeffM2501** [comment]
> > @JeffM2501 @Peter0x44 Just noticed there are no operators for `Matrix`, should there be some added?


I have made another PR for matrix operators and merged back the formatting changes.

**Comment by RobLoach** [comment]
Oh, this is :fire::fire::fire::fire::fire::fire:

Feel free to move anything else in too :wink:

**Pull request merged**